cmd/thanos/compact.go

```diff
@@ -480,6 +490,12 @@ func (cc *compactConfig) registerFlag(cmd *kingpin.CmdClause) *compactConfig {
 		"as querying long time ranges without non-downsampled data is not efficient and useful e.g it is not possible to render all samples for a human eye anyway").
 		Default("false").BoolVar(&cc.disableDownsampling)
 
+	cmd.Flag("compaction.disable", "Disables compaction. This is not recommended "+
```

Comment:
why are we deploying compactor without compaction? can you describe the usecase

Reply:
As stated in the PR description. I'm using Swift and it's eventually consistent.
My concern is that running for example downsampling right after compaction could lead to higher chance of consistency issues. 
This way I could run compaction, downsampling and retention separately with some delay.
So definitely not disabling compaction just kind of extending variability of the deployment model.   
But I know it's weird use case probably, I'd understand if this would be rejected :)

Reply:
Hmm I'm not against this then. Let's see what others think.